lightning/src/chain/chainmonitor.rs

```diff
@@ -42,6 +41,57 @@ use prelude::*;
 use sync::RwLock;
 use core::ops::Deref;
 
+/// `Persist` defines behavior for persisting channel monitors: this could mean
+/// writing once to disk, and/or uploading to one or more backup services.
+///
+/// Note that for every new monitor, you **must** persist the new `ChannelMonitor`
+/// to disk/backups. And, on every update, you **must** persist either the
+/// `ChannelMonitorUpdate` or the updated monitor itself. Otherwise, there is risk
+/// of situations such as revoking a transaction, then crashing before this
+/// revocation can be persisted, then unintentionally broadcasting a revoked
+/// transaction and losing money. This is a risk because previous channel states
+/// are toxic, so it's important that whatever channel state is persisted is
+/// kept up-to-date.
+pub trait Persist<ChannelSigner: Sign> {
```

Comment:
Do we expect a watchtower implementation to persist `ChannelMonitor`s independently of a `ChainMonitor` and reuse a `Persist` implementation to do so? That could be an argument of keeping this outside of `chainmonitor.rs`.

Reply:
Chatted about it offline, but I don't think so - I'd think they'd use `chain::Watch` directly and never think about `Persister`, especially after #1108.